pkg/trace/writer/trace.go

```diff
@@ -216,8 +216,26 @@ func (w *TraceWriter) flush() {
 		gzipw.Write(b)
 		gzipw.Close()
 
-		for _, sender := range w.senders {
-			sender.Push(p)
+		var payloads []*payload
+		if len(w.senders) == 1 {
+			// Avoid an allocation when only one endpoint is configured.
+			payloadsArr := [1]*payload{p}
+			payloads = payloadsArr[:]
+		} else {
+			// If there is more than one sender then we need to create a payload clone for each
+			// one because pooling is managed at the per-sender level so we need separate payloads
+			// for each sender to avoid double-put pooling bugs.
+			payloads = make([]*payload, 0, len(w.senders))
+			for i := range w.senders {
+				p := p
+				if i != 0 {
+					p = p.clone()
+				}
+				payloads = append(payloads, p)
+			}
+		}
+		for i, sender := range w.senders {
```

Comment:
After applying the above changes, both the `payloads` declaration and this `for` loop can go into the `else` branch.